### PR TITLE
refactor: convert io.null and io.undefined to ioNullOrUndefined [DET-3747]

### PR DESCRIPTION
**Pull request by hkang1**
## Description

App breaks when the io-ts decoder encounters an unexpected `null` or `undefined` json property value. Apply the io union type of `null` or `undefined` to all properties that is defined as `null` or `undefined` so we don't have to guess which might come through.

## Commentary (optional)

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: webui/react/src/services/decoder.ts [hunk 1]
```diff
@@ -161,7 +161,7 @@ export const jsonToExperiment = (data: unknown): Experiment => {
     endTime: io.end_time || undefined,
     id: io.id,
     ownerId: io.owner_id,
-    progress: io.progress !== null ? io.progress : undefined,
+    progress: io.progress || undefined,
```
Comment (hamidzr) [inline webui/react/src/services/decoder.ts:162]: could progress be 0?

File: webui/react/src/services/decoder.ts [hunk 2]
```diff
@@ -247,7 +247,7 @@ export const jsonToExperimentDetails = (data: unknown): ExperimentDetails => {
     endTime: ioType.end_time || undefined,
     id: ioType.id,
     ownerId: ioType.owner.id,
-    progress: ioType.progress !== null ? ioType.progress : undefined,
+    progress: ioType.progress || undefined,
```
Comment (hamidzr) [inline webui/react/src/services/decoder.ts]: ditto

File: webui/react/src/ioTypes.ts [hunk 3]
```diff
@@ -21,6 +21,8 @@ export const decode = <T>(type: io.Mixed, data: any): T => {
   }
 };
 
+const ioNullUndefined = io.union([ io.null, io.undefined ]);
```
[the remark this answers is not in the document]
Comment (hkang1) [reply webui/react/src/ioTypes.ts:24]: oh good catch, I just copied it from your PR (hoping to minimize conflicts) and didn't realize that it was even different.

File: webui/react/src/services/decoder.ts (hunk 1 above)
Comment (hkang1) [reply webui/react/src/services/decoder.ts:162]: ah yes, another great catch!

**Pull request merged**